### ISSUE TITLE
Organize GraphFrames Hackathon

### ISSUE THREAD

**Issue by rjurney** [graphframes/graphframes#460]
I am going to organize a GraphFrames hackathon. Sign up for issues here.

* I am going to work on some outstanding bugs in GraphX to engage with the Spark 4 GraphX branch
* @decius has expressed interest in a Louvain implementation #299

Anyone else?

**Comment by SauronShepherd** [comment]
I'm all in! Count on me to fix bugs!

**Comment by SauronShepherd** [comment]
What about including OpenCypher as query language? Maybe for... Spark 5 or so

**Comment by Knorreman** [comment]
I am interested. What would a hackathon in this context entail?

**Comment by rjurney** [comment]
> I am interested. What would a hackathon in this context entail?

We establish a day where we all get together and all day hack on a particular task alone or in pairs. Sometimes people team up remotely on issues, which is fun! We hang out in a common chat / video room to support one another and have camaraderie as we work. It rocks!

**Comment by rjurney** [comment]
> I'm all in! Count on me to fix bugs!

Awesome!

**Comment by james-willis** [comment]
Does this package have any active maintainers? It seems like PRs are not being accepted in this repo. It would be a shame to have a hackathon to write a bunch of code that is never merged.

**Comment by rjurney** [comment]
> Does this package have any active maintainers? It seems like PRs are not being accepted in this repo. It would be a shame to have a hackathon to write a bunch of code that is never merged. 

Yes, we cut a release not that long ago.

**Comment by rjurney** [comment]
> Does this package have any active maintainers? It seems like PRs are not being accepted in this repo. It would be a shame to have a hackathon to write a bunch of code that is never merged.

James, I now have maintainer permissions :)

**Comment by rjurney** [comment]
The time has come... I am going to plan and announce this. Need to pick a date... I guess I will choose a Saturday?

**Comment by SauronShepherd** [comment]
GREAT! Any timezone in mind? I live in Spain, for example, and we have CET
(GMT+1), but I usually wake up at 5am, even Saturdays 😅, so ...).

**Comment by rjurney** [comment]
I propose Friday, February 21, from 8AM EST to 5PM PST (8PM EST) or 2PM CET until 2AM CET… if anyone (@SauronShepherd ) wants to host the Google Hangout earlier than 2PM CET let me know and we can start at 8AM CET :) basically almost 24 hours spanning time zones!

**Comment by SauronShepherd** [comment]
Hmmm... I'd have to take that day off from work, but sounds really
tempting! Let's do it—count me in!

Just to clarify, are you suggesting we combine both activities into the
same event—solving technical difficulties while setting up the environments
and the hackathon itself?

**Comment by rjurney** [comment]
Yes, I think we should pin "get help with setting up environments" for a
certain time in CET and US time, maybe 10AM CET (1AM PST) and 11AM EST (5PM
CET)? If I see you run through some fixes I can do that if you're off by
then.

So we would announce:

Hackathon!

* Get help setting up your environment with a presentation and help session
held at one of two times!
    * 10AM CET - Angel Pascua will assist with project builds at the CLI
and with IntelliJ
    * 11AM EST / 8AM PST - Russell Jurney will assist with project builds
at the CLI and with IntelliJ

Come out and meet and get support... blah blah

What do you think?

Russell

**Comment by rjurney** [comment]
Hello ticket followers: the time to announce the Hackathon is here! This just went out on the google group and Spark user list :) See you on Friday, February 21st!

-------------------------------------------------------------------

You are being contacted because you are on the Spark user or GraphFrames user list or have expressed interest in a GraphFrames Hackathon.

The first annual GraphFrames Hackathon is taking place on Friday, February 21 :) As we have had interest from the US and Europe, we are going to hack round the clock. You are invited to attend starting at 8AM CET (middle of the night for Americans: 2AM EST/11PM PST) all the way through 5PM PST (2AM CET).

If you need something to work on, file your idea or look for ["help wanted" tags on issues](https://github.com/graphframes/graphframes/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22help%20wanted%22). If you have an idea, please [file an issue!](https://github.com/graphframes/graphframes/issues/new?template=Blank+issue) We look forward to seeing you there!

**Comment by SemyonSinchenko** [comment]
@rjurney is it done? Can we close it?
